hubmap/query_app/apps.py

```diff
@@ -73,9 +108,27 @@ def attempt_to_open_file(file_path, key=None):
         assert file_path.suffix == ".hdf5"
         try:
             df = pd.read_hdf(file_path, key)
+
+            columns_dict = {"percentages": ["var_id", "cutoff", "dataset"], "cell": ["dataset"]}
+
+            if key == "cell":
+                if "clusters" in df.columns:
+                    df = df[~df["clusters"].isna()]
+
+                for i in df.index:
```

Comment:
It might be more efficient to do this with something like https://pandas.pydata.org/docs/reference/api/pandas.Series.str.split.html -- it's more efficient to do this iteration in lower-level code.